test/gtest/layout_transpose.cpp

```diff
@@ -260,6 +249,20 @@ struct LayoutTransposeTest_2D : public ::testing::TestWithParam<std::tuple<uint3
     uint32_t n;
     uint32_t c;
 
+    miopen::Invoker PrepareLayoutTransposeInvoker(const miopen::ExecutionContext& ctx_,
+                                                  miopenDataType_t data_type_,
+                                                  uint32_t n_,
+                                                  uint32_t c_,
+                                                  uint32_t h_,
+                                                  uint32_t w_)
+    {
+        TRANSPOSE_SOL transpose_sol(ctx_, data_type_, n_, c_, h_, w_);
```

Comment:
same here. This should be a method on the `BatchedTransposeSolution` class.

Reply:
I actually had it there initially, but I moved it because it is specific to testing and other Solvers do not provide such helpers in their public API.

In the past, I have seen such test helpers placed into a friend class. I don't have strong feelings any way, but this does the simplest and better from a consistency standpoint. Of course, the point is improvement, so consistency is not a big motivator. It does add a couple orders of magnitude to the scope of the suggestion though.

Reply:
I think anyone using the `BatchedTransposeSolution` class would want this method and it would avoid repetition as seen in your own test case.